### PR TITLE
Added gRPC service call metrics

### PR DESCRIPTION
**Pull request by mfateev**

### CODE REVIEW

File: src/test/java/io/temporal/workflow/MetricsTest.java [hunk 1]
```diff
@@ -335,4 +354,63 @@ public WorkflowInvoker interceptExecuteWorkflow(
...
+    }
+
+    // Wait for reporter
+    Thread.sleep(600);
+
+    verify(reporter, times(1))
+        .reportCounter("temporal-StartWorkflowExecution.temporal-request", new HashMap<>(), 1);
+    verify(reporter, times(1))
+        .reportCounter(
+            "temporal-StartWorkflowExecution.temporal-invalid-request", new HashMap<>(), 1);
```
Comment (shawnhathaway) [inline src/test/java/io/temporal/workflow/MetricsTest.java:413]: Are these defined somewhere as consts or is there a common way to generate them?

File: src/test/java/io/temporal/workflow/MetricsTest.java [hunk 2]
```diff
@@ -335,4 +354,63 @@ public WorkflowInvoker interceptExecuteWorkflow(
...
+          testEnvironment.getWorkflowClient().getWorkflowServiceStubs();
+
+      serviceStubs.blockingStub().describeNamespace(DescribeNamespaceRequest.newBuilder().build());
+      fail("failure expected");
+    } catch (StatusRuntimeException e) {
+      assertEquals(Status.Code.UNIMPLEMENTED, e.getStatus().getCode());
+    }
+
+    // Wait for reporter
+    Thread.sleep(600);
```
Comment (shawnhathaway) [inline src/test/java/io/temporal/workflow/MetricsTest.java:377]: Where does 600 come from? Is this long enough for most/all developers who will run these?

Comment (mfateev) [reply src/test/java/io/temporal/workflow/MetricsTest.java:377]: Changed to more reasonable value.

File: src/test/java/io/temporal/workflow/MetricsTest.java (hunk 1 above)
Comment (mfateev) [reply src/test/java/io/temporal/workflow/MetricsTest.java:413]: Replaced with constants

File: src/test/java/io/temporal/workflow/MetricsTest.java (hunk 2 above)
Comment (mfateev) [reply src/test/java/io/temporal/workflow/MetricsTest.java]: It looks like a smaller value leads to intermittent test failures. So keeping it at 600.

**Pull request merged**